pkg/cli/admin/release/mirror.go

```diff
@@ -442,18 +450,64 @@ func (o *MirrorOptions) Run() error {
 	} else {
 		fmt.Fprintf(o.Out, "\nSuccess\nUpdate image:  %s\nMirrored to: %s\n", to, o.To)
 	}
+
+	printInstallInstructions(o.Out, o.From, o.To, repositories)
 	return nil
 }
 
-func sourceImageRef(is *imagev1.ImageStream, name string) (string, bool) {
-	for _, tag := range is.Spec.Tags {
-		if tag.Name != name {
-			continue
-		}
-		if tag.From == nil || tag.From.Kind != "DockerImage" {
-			return "", false
-		}
-		return tag.From.Name, true
+// printInstallInstructions provides detail to the user for using the new mirror for installs.
+// The instructions include yaml formatted output which should be added to the install-config.yaml.
+//
+// Exmaple output:
+//
+//  <descriptive installer usage text>
+//  imageContentSources:
+//  - sources:
+//    - mycompany.com/myrepository/repo                 # destination (mirror) listed first
+//    - registry.svc.ci.openshift.org/ocp/release       # source listed second
+//    - quay.io/openshift-release-dev/ocp-v4.0-art-dev  # repos from image-references
+//
+func printInstallInstructions(out io.Writer, from, to string, repos map[string]struct{}) error {
+	type installConfigSubsectionImageContentSources struct {
+		Sources []string `json:"sources"`
+	}
+	type installConfigSubsection struct {
+		ImageContentSources []installConfigSubsectionImageContentSources `json:"imageContentSources"`
```

Comment:
This needs updating for https://github.com/openshift/api/pull/384 .

Reply:
Can you link to an example ICSP?

Reply:
Not yet, openshift/installer will need to update for that API change first (updating https://github.com/openshift/installer/pull/1910 ) and define the new type.

Reply:
> Not yet, openshift/installer will need to update for that API change first

This is happening in https://github.com/openshift/installer/pull/2048 .

Reply:
Updated.